pkg/imgpkg/registry/keychain_test.go

```diff
@@ -169,6 +170,58 @@ func TestAuthProvidedViaEnvVars(t *testing.T) {
 		}), auth)
 	})
 
+	for i, validHostname := range []string{
+		"localhost:9999",
+		"http://localhost:9999",
+		"https://localhost:9999",
+		"localhost:9999/v1/",
+		"localhost:9999/v2/",
+		"*:9999/v2/",
+		"local*:9999/v2/",
+		"localhost:9999/imgpkg_test",
```

Comment:
do we allow globs within path?

Reply:
No thats a limitation with the package (k8s-pkg-credentialprovider) we are using

https://github.com/vmware-tanzu/carvel-imgpkg/blob/3a779ca0886ab989a22c6928a1eb58cd36dc0bbd/vendor/github.com/vdemeester/k8s-pkg-credentialprovider/keyring.go#L203

Reply:
ah interesting, i guess thats ok for now. could always expand later. what happens if somebody does throw in a star into a path? error, or something else?

Reply:
not an error. it won't match (it will return an Anonymous) and will try the next keychain (iaas, cli, config.json etc)